rust/xaynet-analytics/src/database/controller_data/data_model.rs

```diff
@@ -30,3 +30,33 @@ impl Into<ControllerDataAdapter> for ControllerData {
         ControllerDataAdapter::new(self.time_data_sent.to_rfc3339())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_controller_data_try_from_adapter() {
+        let timestamp_str = "2021-01-01T01:01:00+00:00";
+        let timestamp_parsed = DateTime::parse_from_rfc3339(timestamp_str)
+            .unwrap()
+            .with_timezone(&Utc);
+        let controller_data = ControllerData::new(timestamp_parsed);
+        let adapter = ControllerDataAdapter::new(timestamp_str);
+        assert_eq!(ControllerData::try_from(adapter).unwrap(), controller_data);
+    }
+
+    #[test]
+    fn test_adapter_try_into_controller_data() {
```

Comment:
```suggestion
    fn test_adapter_into_controller_data() {
```
similarly in other data models

Reply:
yup, thanks. will change